### ISSUE TITLE
Concurrent problems in default aggregator module

### ISSUE THREAD

**Issue by dxsup** [Kindling-project/kindling#194]
**Describe the bug**
The result value calculated by the default aggregator is wrong when aggregating and dumping are executed concurrently.  

**How to reproduce?**
I can always reproduce this problem by running the following test case.
```go

func TestConcurrentAggregator(t *testing.T) {
	aggKindMap := &AggregatedConfig{KindMap: map[string][]KindConfig{
		"duration": {
			{Kind: SumKind, OutputName: "duration_sum"},
			{Kind: CountKind, OutputName: "request_count"},
		},
	}}

	aggregator := NewDefaultAggregator(aggKindMap)
	labels := model.NewAttributeMap()
	labels.AddStringValue("key", "value")

	labelSelectors := internal.NewLabelSelectors(
		internal.LabelSelector{Name: "key", VType: internal.StringType},
	)

	wg := sync.WaitGroup{}
	wg.Add(1)
	stopCh := make(chan bool)

	var runLoop = 100000
	var duration int64 = 100
	go func() {
		for i := 0; i < runLoop; i++ {
			gaugeValues := []*model.Gauge{
				{"duration", duration},
			}
			gaugeGroup := model.NewGaugeGroup("testGauge", labels, 0, gaugeValues...)
			aggregator.Aggregate(gaugeGroup, labelSelectors)
			time.Sleep(time.Microsecond)
		}
		stopCh <- true
		wg.Done()
	}()

	wg.Add(1)
	var durationSum int64
	var requestCount int64
	go func() {
		ticker := time.NewTicker(500 * time.Microsecond)
		for {
			select {
			case <-ticker.C:
				ret := aggregator.Dump()
				for _, g := range ret {
					for _, v := range g.Values {
						if v.Name == "duration_sum" {
							durationSum += v.Value
						} else if v.Name == "request_count" {
							requestCount += v.Value
						}
					}
				}
			case <-stopCh:
				ret := aggregator.Dump()
				for _, g := range ret {
					for _, v := range g.Values {
						if v.Name == "duration_sum" {
							durationSum += v.Value
						} else if v.Name == "request_count" {
							requestCount += v.Value
						}
					}
				}
				wg.Done()
				return
			}
		}
	}()
	wg.Wait()
	expectedDurationSum := int64(runLoop) * duration
	expectedRequestCount := int64(runLoop)
	if durationSum != expectedDurationSum || requestCount != expectedRequestCount {
		t.Errorf("request_count = %v, duration_sum := %v", requestCount, durationSum)
	}
}
```

**What did you expect to see?**
The result value should be as expected in the above test case.

**What did you see instead?**
=== RUN   TestConcurrentAggregator
    default_aggregator_test.go:84: request_count: exptected 100000, got 99545
    default_aggregator_test.go:85: duration_sum: expected 10000000, got 9954700
--- FAIL: TestConcurrentAggregator (0.93s)

FAIL

**Comment by dxsup** [comment]
This problem happens when the executions of `default_aggregator` run in the following order.
1. Get a recorder when calling the `Aggregate` method;
2. The same recorder is reset when calling the `Dump` method;
3. This recorder begins to `Record` a value.

In this case, the recorder can't be put into the map again, so the value it has calculated is missing. To fix this issue, the recorders can't be loaded from the map when the `Dump` method is called.

**Comment by dxsup** [comment]
Closed via #195
